src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/azure_devops_build_iteractive.py

```diff
@@ -259,37 +456,95 @@ def process_remote_repository(self):
+            self.release_definition_name = "_release_github_" + self.github_repository.replace("/", "_", 1)[0:256]
+
+    def process_github_service_endpoint(self):
+        service_endpoints = self.adbp.get_github_service_endpoints(
+            self.organization_name, self.project_name, self.github_repository
+        )
+
+        if not service_endpoints:
+            service_endpoint = self.adbp.create_github_service_endpoint(
+                self.organization_name, self.project_name, self.github_repository, self.github_pat
+            )
+        else:
+            service_endpoint = service_endpoints[0]
+            self.logger.warning("Detected a Github service endpoint already exists: {name}".format(
+                                name=service_endpoint.name))
+
+        self.github_service_endpoint_name = service_endpoint.name
 
-    def process_service_endpoint(self):
+    def process_functionapp_service_endpoint(self, scenario):
+        repository = self.repository_name if scenario == "AZURE_DEVOPS" else self.github_repository
         service_endpoints = self.adbp.get_service_endpoints(
-            self.organization_name, self.project_name, self.repository_name
+            self.organization_name, self.project_name, repository
         )
 
         # If there is no matching service endpoint, we need to create a new one
         if not service_endpoints:
             try:
                 service_endpoint = self.adbp.create_service_endpoint(
-                    self.organization_name, self.project_name, self.repository_name
+                    self.organization_name, self.project_name, repository
                 )
             except RoleAssignmentException:
-                self.adbp.remove_git_remote(self.organization_name, self.project_name, self.repository_name)
-                raise CLIError("To use the Azure DevOps Pipeline Build,{ls}"
-                               "We need to assign a contributor role to the Azure Functions release service principle.{ls}"
-                               "Please ensure you are the owner of the subscription, or have role assignment write permission.".format(ls=os.linesep))
+                if scenario == "AZURE_DEVOPS":
+                    self.adbp.remove_git_remote(self.organization_name, self.project_name, repository)
+                raise CLIError("{ls}To create a build through Azure Pipelines,{ls}"
+                               "The command will assign a contributor role to the "
+                               "Azure function app release service principle.{ls}"
+                               "Please ensure that:{ls}"
+                               "1. You are the owner of the subscription, "
+                               "or have roleAssignments/write permission.{ls}"
+                               "2. You can perform app registration in https://portal.azure.com/#blade/"
```

Comment:
Will the link still work in sovereign clouds?

Reply:
Good catch, thanks for pointing this out!

Reply:
Remove the url, since the url is not the only way for user to check Active Directory application registration, we should report the issue in a higher level view.